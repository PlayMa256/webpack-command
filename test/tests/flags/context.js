const { apply, build, test, validate } = require('../../util');

const fixture = 'context/context';
const opts = { fixture };

let config;

test('--context', module, () => {
  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should build`, () =>
    build(config).then((result) => {
      expect(result).toMatchSnapshot();
    }));
});
