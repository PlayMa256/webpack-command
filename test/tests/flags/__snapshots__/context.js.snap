// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --context > should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>/test/fixtures/common",
  "entry": "./entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --context > should build #0 1`] = `
"Version: webpack 4.12.1
  Asset     Size  Chunks             Chunk Names
main.js  3.8 KiB    main  [emitted]  main
Entrypoint main = main.js
[./entry-a.js] 41 bytes {main} [built]"
`;
